File: backend/src/common/providers/hotel.provider.ts
import { ProviderResult } from './provider.result';

export interface HotelProvider {
  providerId: string;
  providerName: string;
  searchHotels(params: {
    city: string;
    checkIn: Date;
    checkOut: Date;
    guests?: number;
  }): Promise<ProviderResult<any[]>>;
}

File: lib/core/providers/contracts/hotel_provider.dart
import '../../../core/domain/models/offers/hotel_offer.dart';

abstract interface class HotelProvider {
  String get providerId;
  String get providerName;

  Future<List<HotelOffer>> searchHotels({
    required String city,
    required DateTime checkIn,
    required DateTime checkOut,
    int? guests,
  });
}